Project2015To2017/Reading/SolutionReader.cs

```diff
@@ -11,35 +11,74 @@ public sealed class SolutionReader
 	{
 		public static readonly SolutionReader Instance = new SolutionReader();
 
+		// An example of a project line looks like this:
+		//  Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClassLibrary1", "ClassLibrary1\ClassLibrary1.csproj", "{05A5AD00-71B5-4612-AF2F-9EA9121C4111}"
+		private static readonly Lazy<Regex> crackProjectLine = new Lazy<Regex>(
```

Comment:
I still believe `Lazy` is a good idea. The overhead is actually small (I looked at `Lazy` sources), but improves cases when there is no need to **compile** heavy Regex, like when only projects are specified.

Reply:
I think in those cases it won't ever initialise SolutionReader including the static fields, but could be wrong

Reply:
I'm afraid we can't guarantee that. The first time `Instance` or any other field is accessed, all static fields are initialized. Imagine we add sometime later a new static field to SolutionReader and access that in ProjectConverter unconditionally, then regex will always be initialized and compiled. MSBuild folks didn't do it out of pure will to have _perfect_ code.